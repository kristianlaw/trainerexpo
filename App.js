import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { NavigationContainer} from '@react-navigation/native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import Progress from './pages/Progress';
import Trainingplan from './pages/Trainingplan';
import Home from './pages/Home';
import Startscreen from './pages/Startscreen';


const Tab = createBottomTabNavigator();

export default function App() {
  return (

    <NavigationContainer>
        <Tab.Navigator>
        <Tab.Screen name="Home" component={Home} />
        <Tab.Screen name="Startscreen" component={Startscreen} />
        <Tab.Screen name="Progress" component={Progress} />
        <Tab.Screen name="Trainingplan" component={Trainingplan} />
        </Tab.Navigator>
    </NavigationContainer>

  );
}
