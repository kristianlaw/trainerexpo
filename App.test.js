import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
//Source: https://docs.expo.io/guides/testing-with-jest/
//Uses Jest to run test cases
//npm run test



//Our test will be the expected state of the <App /> to have 1 child element:
 describe('<App />', () => {
  it('1 child within', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(1);
  });
});

//etsii tekstin App.js tiedostosta
//teksti minkä etsii niin on "Development"
let etsiTeksti = function(render, element){
    return true;
}
it('finds text in App.js', ()=>{
   let render = renderer.create(
       <App />).toJSON();
   expect(etsiTeksti(render, 'development')).toBeDefined();
})

//Snapshot testing
//Testien valitettaessa "1 snapshot obsolete" --> 'npm test -- -u'
//Tämä poistaa obsolete snapshotin
test('proper rendering', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
