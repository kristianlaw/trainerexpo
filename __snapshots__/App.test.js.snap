// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proper rendering 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Development-branch!
  </Text>
</View>
`;
